src/main/java/de/tum/in/www1/artemis/service/AutomaticTextFeedbackService.java

```diff
@@ -0,0 +1,66 @@
+package de.tum.in.www1.artemis.service;
+
+import static java.util.Comparator.comparing;
+import static java.util.stream.Collectors.toList;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import org.jetbrains.annotations.NotNull;
+import org.springframework.context.annotation.Profile;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import de.tum.in.www1.artemis.domain.*;
+import de.tum.in.www1.artemis.domain.enumeration.FeedbackType;
+
+@Service
+@Profile("automaticText")
+public class AutomaticTextFeedbackService {
+
+    private final FeedbackService feedbackService;
+
+    private static final double DISTANCE_THRESHOLD = 1;
+
+    public AutomaticTextFeedbackService(FeedbackService feedbackService) {
+        this.feedbackService = feedbackService;
+    }
+
+    /**
+     * @param result
+     */
+    @Transactional(readOnly = true)
+    public void suggestFeedback(@NotNull Result result) {
```

Comment:
This seems like more complicated logic in here.
So I think better JavaDoc is a must and also this should have (a couple of) Unit tests for different scenarios.

Reply:
✅